### PR TITLE
OSDOCS#8217: updating Agent doc MCE channel number

### PR DESCRIPTION
**Pull request by skopacz1**
[OSDOCS-8217](https://issues.redhat.com/browse/OSDOCS-8217)

Version(s): 4.13+

This PR updates the channel for the MCE Operator in on of the Agent docs' sample files. It also cleans up some other parts of the doc.

QE review:
- [x] QE has approved this change.

Preview: [Example `mce_subscription.yaml`](https://67878--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_with_agent_based_installer/preparing-an-agent-based-installed-cluster-for-mce#preparing-an-inital-cluster-deployment-for-mce-connected_preparing-an-agent-based-installed-cluster-for-mce:~:text=Example%20mce_subscription.yaml)

### CODE REVIEW

File: modules/preparing-an-inital-cluster-deployment-for-mce-connected.adoc [hunk 1]
```diff
@@ -57,7 +57,7 @@ The installer does not validate extra manifests.
     name: multicluster-engine
     namespace: multicluster-engine
   spec:
-    channel: "stable-2.1"
+    channel: "stable-2.3"
```
Comment (skopacz1) [inline modules/preparing-an-inital-cluster-deployment-for-mce-connected.adoc:60]: This change is based on [this bundle](https://catalog.redhat.com/software/containers/multicluster-engine/mce-operator-bundle/6160406290fb938ecf6009c6?architecture=amd64&image=6537e7c0fdfcb606c55e382f&container-tabs=dockerfile), where lines 8 and 14 suggest that the `stable-2.3` channel is valid for versions 4.13 and 4.14.

I'll likely need to review this example yaml again for the 4.15 docs.

**Comment by skopacz1** [comment]
@mhanss could you PTAL when you get the chance? Thanks!

**Comment by mhanss** [comment]
LGTM

**Pull request merged**